sqladmin/models.py

```diff
@@ -1179,6 +1182,32 @@ async def generate(writer: Writer) -> AsyncGenerator[Any, None]:
             headers={"Content-Disposition": f"attachment;filename={filename}"},
         )
 
+    async def _export_json(
+        self,
+        data: List[Any],
+    ) -> StreamingResponse:
+        async def generate() -> AsyncGenerator[str, None]:
+            yield "["
+            first_row = True
+            for row in data:
+                if not first_row:
+                    yield ","
+                else:
+                    first_row = False
+                row_dict = {
+                    name: await self.get_prop_value(row, name)
+                    for name in self._export_prop_names
+                }
+                yield json.dumps(row_dict)
```

Comment:
Can we check length of data before loop and if there is more than one item set `separator=","` then return `yield json.dumps() + separator`
I want to avoid this branching in loop.

Reply:
dont get it - smth linke that?

`async def generate() -> AsyncGenerator[str, None]:
        yield "["
        separator = "," if len(data) > 1 else ""
        first_row = True
        
        for row in data:
            if not first_row:
                yield separator
            else:
                first_row = False
            
            row_dict = {
                name: await self.get_prop_value(row, name)
                for name in self._export_prop_names
            }
            yield json.dumps(row_dict)
        
        yield "]"
`

Reply:
oh i got it, now looks like you want

Reply:
@aminalaee is it ok? i want to use it on my side asap) can we merge?